oauth2/__init__.py

```diff
@@ -484,7 +484,7 @@ def get_normalized_parameters(self):
     def sign_request(self, signature_method, consumer, token):
         """Set the signature parameter to the result of sign."""
 
-        if not self.is_form_encoded:
+        if (not self.is_form_encoded) and self.body is not None:
```

Comment:
Should this be just `and self.body`? This will still encode `body = ""`; not sure if that's expected given your intent here.